src/Serilog.Sinks.Datadog.Logs/Sinks/Datadog/DatadogHttpClient.cs

```diff
@@ -47,6 +47,14 @@ public DatadogHttpClient(DatadogConfiguration config, DatadogLogRenderer rendere
             _url = $"{config.Url}/api/v2/logs";
             _renderer = renderer;
         }
+        
+        protected DatadogHttpClient(DatadogConfiguration config, HttpClient client, string url, DatadogLogRenderer renderer)
```

Comment:
This is a good suggestion.
Instead of adding a new constructor - can we replace the main constructor with the signature you proposed?

We can encapsulate the HttpClient setup logic either in a helper function or in a subclass of `HttpClient` (outside of `DatadogHttpClient`) thus removing the need to pass the API key in directly.

I think this should accomplish the same thing without having multiple constructors.

Reply:
I have created DatadogHttpClientBase with all important things except constructor logic. It is now inside DatadogHttpClient.
Is that ok, or should I change it in other way?

Reply:
I was thinking - instead of a base class - we could move the `HttpClient` implementation into it's own class. Something like: 

```C#
public class DatadogApiClient: HttpClient {
    private const string _version = "0.4.0";
    public DatadogApiClient(string apiKey): base()
    {
        DefaultRequestHeaders.Add("DD-API-KEY", apiKey);
        DefaultRequestHeaders.Add("DD-EVP-ORIGIN", "Serilog.Sinks.Datadog.Logs");
        DefaultRequestHeaders.Add("DD-EVP-ORIGIN-VERSION", _version);
    }
}
```

And then update the constructor: 

```C#
public DatadogHttpClient(DatadogConfiguration config, DatadogLogRenderer renderer, HttpClient client)
{
    _config = config;
    _client = client;
    _url = $"{config.Url}/api/v2/logs";
    _renderer = renderer;
}
```

This way you can either pass in the default implementation of the HTTP Client (with the API key headers) - or your own version. Can also pass in the URL though the constructor (like in your first commit).

Reply:
Ability to manipulate URL is crucial here, as for implementing this for Blazor, we need to put API Key as query parameter (`$"{config.Url}/api/v2/logs?ddsource=browser&dd-api-key={apiKey}"`).
Writing own HttpClient that appends query parameters doesn't seem like a good idea, especially given that some classes used to manipulate URL are not available in .NET Standard 1.3. and .NET Framework versions supported.

Passing URL through constructor would mean that creating this URL would need to be moved to `DatadogSink.CreateDatadogClient`. If that is ok, then I can do it like that.

Reply:
> Passing URL through constructor would mean that creating this URL would need to be moved to DatadogSink.CreateDatadogClient. If that is ok, then I can do it like that.


👍 Yep - this sounds fine to me.